samza-core/src/main/java/org/apache/samza/operators/impl/ControlMessageSender.java

```diff
@@ -43,9 +47,14 @@
   }
 
   void send(ControlMessage message, SystemStream systemStream, MessageCollector collector) {
-    SystemStreamMetadata metadata = metadataCache.getSystemStreamMetadata(systemStream, true);
-    int partitionCount = metadata.getSystemStreamPartitionMetadata().size();
-    LOG.info(String.format("Broadcast %s message from task %s to %s with %s partition",
+    Integer partitionCount = partitions.get(systemStream);
```

Comment:
Simplify this by using .putIfAbsent / computeIfAbsent?

Reply:
+1

Reply:
Good suggestion. computeIfAbsent works pretty nicely here.